genesis/engine/solvers/rigid/gjk_decomp.py

```diff
@@ -0,0 +1,3394 @@
+import taichi as ti
+import genesis as gs
+import genesis.utils.geom as gu
+from .support_decomp import Support
+
+EPA_P2_NONCONVEX = 2
+EPA_P2_FALLBACK3 = 3
+EPA_P3_BAD_NORMAL = 4
+EPA_P3_INVALID_V4 = 5
+EPA_P3_INVALID_V5 = 6
+EPA_P3_MISSING_ORIGIN = 7
+EPA_P3_ORIGIN_ON_FACE = 8
+EPA_P4_MISSING_ORIGIN = 9
+EPA_P4_FALLBACK3 = 10
+
+
+@ti.data_oriented
+class GJK:
+    def __init__(self, rigid_solver):
+        self._solver = rigid_solver
+        self._B = rigid_solver._B
+
+        self.FLOAT_MIN = gs.np_float(1e-15)
+        self.FLOAT_MIN_SQ = self.FLOAT_MIN * self.FLOAT_MIN
+        self.FLOAT_MAX = gs.np_float(1e15)
+        self.FLOAT_MAX_SQ = self.FLOAT_MAX * self.FLOAT_MAX
+        self.tolerance = gs.np_float(1e-6)
+        self.collision_eps = gs.np_float(1e-6)
+
+        # Maximum number of contact points to find per pair
+        self.max_contacts_per_pair = 50
```

Comment:
This is way too large no? How many contact points may be computed by GJK?! I'm not expecting more than something like 8 contact points per pair. We should avoid excessive memory allocation if possible.

Reply:
In multi-contact detection algorithm, we first detect two parallel faces, each of which is from the colliding objects. Then, we run polygon clipping algorithm for those faces to find the intersection polygon, as shown below. Then the vertices of the intersection polygon become the contact points. So if we only consider usual cases where the faces are triangles or quads, 8 would be enough, but I think Mujoco used 50 to consider harder cases, too (like the case shown below). It was also for the compatibility, I'll use the compatibility flag here too.
![image](https://github.com/user-attachments/assets/8025765f-10e9-4ef4-959b-c212a6df8c30)

Reply:
> So if we only consider usual cases where the faces are triangles or quads, 8 would be enough, but I think Mujoco used 50 to consider harder cases, too (like the case shown below)

Yeah we can stick with triangles and quads definitely! That would be save a lot of GPU memory :)